core/modules/xrdsvc/StreamBuffer.cc

```diff
@@ -38,17 +38,42 @@ LOG_LOGGER _log = LOG_GET("lsst.qserv.xrdsvc.StreamBuffer");
 }
 
 
+using namespace std;
+
+
 namespace lsst {
 namespace qserv {
 namespace xrdsvc {
 
-std::atomic<size_t> StreamBuffer::_totalBytes(0);
+
+atomic<int64_t> StreamBuffer::_maxTotalBytes{40'000'000'000};
+atomic<int64_t> StreamBuffer::_totalBytes(0);
+mutex StreamBuffer::_createMtx;
+condition_variable StreamBuffer::_createCv;
+
+
+static void StreamBuffer::setMaxTotalBytes(int64_t maxBytes) {
+    LOGS(_log, LOG_LVL_INFO, "StreamBuffer::setMaxTotalBytes maxBytes=" << maxBytes);
+    if (maxBytes < 0) {
+        throw invalid_argument("StreamBuffer::setMaxTotalBytes negative " + to_string(maxBytes));
```

Comment:
Another optimization that makes this code a bit more readable/shorter is to define this variable at the beginning of each method:
```
string const context = "StreamBuffer::" + string(__func__) + " ";
```
After that you will just use this in the rest of the method:
```
throw invalid_argument(context + "negative " + to_string(maxBytes));
```
This technique helps with refactoring the names of methods or classes. It also makes exception throwing (and LSST logging) statements  less nosier.